### DISCUSSION TITLE
Runtime Config in composables

### DISCUSSION THREAD

**Discussion by Gyurmatag** [nuxt/nuxt#17812] [Questions] [answered]
Here is my custom fetch in my compasables:
/composables/myFetch.ts

```
import type { FetchOptions } from "ohmyfetch";
import config from "#config";

const baseURL = config.API_BASE;

export const useCustomFetch = (path: string, opts?: FetchOptions) => {
  return $fetch(path, { baseURL, ...(opts && { ...opts }) });
};
```

I'm trying to reference my env variables by this documentation: https://v3.nuxtjs.org/docs/usage/runtime-config/#api-routes
useRuntimeConfig() is not working here, so I looked other ways by importing from '#config'. But it throws me this error:

>  ERROR  Nitro aliases cannot be imported in the Vue part of your app. [importing #config from composables/myFetch.ts]                                                                                                                                                                                        14:01:10

What could be the problem?
Thank you!

**Comment by Diizzayy** [comment, marked as answer]
@Gyurmatag `#config` is only meant to be used in nitro (API Routes for instance)

ensure that your nuxt config has the variables added to `publicRuntimeConfig`

```ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  publicRuntimeConfig: {
    API_BASE: '' // setting `API_BASE` in your `.env` will override this
  }
})
```

After doing so, you'll then be able to access your variables via `useRuntimeConfig()`

```
import type { FetchOptions } from 'ohmyfetch'

export const useCustomFetch = (path: string, opts?: FetchOptions) => {
  const { API_BASE: baseURL } = useRuntimeConfig()

  if (!baseURL) throw new Error('baseURL is not defined')

  return $fetch(path, { baseURL, ...(opts && { ...opts }) })
}
```
